### PR TITLE
Fix needless buffering when running prove with -j

### PR DESCRIPTION
**Pull request by AlexDaniel**
See [RT #132108](https://rt.perl.org/Ticket/Display.html?id=132108).

Submitting this pull request just to get more eyes on it. I'm pretty sure that we can come up with a better solution.

### CODE REVIEW

**Comment by lizmat** [comment]
Why is the buffering needless?  If it is needless, but it enhances performance, why remove it?  If the buffering reduces perfomance, why?  that seems counter-intuitive?

**Comment by niner** [comment]
On Sonntag, 17. September 2017 11:57:04 CEST lizmat wrote:
> Why is the buffering needless?  If it is needless, but it enhances
> performance, why remove it?  If the buffering reduces perfomance, why? 
> that seems counter-intuitive?

Buffering doesn't reduce performance, it increases latency. It keeps the TAP 
from reaching the harness in a timely manner, so the harness cannot print an 
updated status.

**Comment by lizmat** [comment]
Understood (now)  :-)

File: lib/Test.pm6 [hunk 1]
```diff
@@ -39,6 +39,8 @@ my int $done_testing_has_been_run = 0;
 _init_vars();
 
 sub _init_io {
+    nqp::setbuffersizefh(nqp::getstdin(),  0);
```
Comment (jnthn) [inline lib/Test.pm6]: Setting it on STDIN makes no sense; the op is only about output buffering. Setting it on both STDOUT and STDERR might make sense.

Comment (AlexDaniel) [reply lib/Test.pm6]: do'h! That's exactly what I meant…

**Comment by AlexDaniel** [comment]
FWIW Test.pm6 is already using nqp rather heavily.

**Comment by AlexDaniel** [comment]
I've created a separate ticket for Perl 6-level setbuffersize, see [RT #132111](https://rt.perl.org/Ticket/Display.html?id=132111).

**Comment by jnthn** [comment]
Yeah, to be clear I wasn't saying we had to do the "provide a way without nqp ops" as a pre-req for merging this, just that others will likely want to do the same and we should support it more properly. I'll try and get to it sometime in the next weeks, and try and remember to update Test.pm6 to use it also.

**Pull request merged**